docs/proposals/job.md

```diff
@@ -0,0 +1,145 @@
+
+The new `Job` object will have the following content:
+
+```go
+// Job represents the configuration of a single job.
+type Job struct {
+    TypeMeta
+    ObjectMeta
+
+    // Spec is a structure defining the expected behavior of a job.
+    Spec JobSpec
+
+    // Status is a structure describing current status of a job.
+    Status JobStatus
+}
+
+// JobList is a collection of jobs.
+type JobList struct {
+    TypeMeta
+    ListMeta
+
+    Items []Job
+}
+```
+
+`JobSpec` structure is defined to contain all the information how the actual job execution
+will look like.
+
+```go
+// JobSpec describes how the job execution will look like.
+type JobSpec struct {
+
+    // MaxParallelism specifies the maximum desired number of pods the job should
+    // run at any given time. The actual number of pods running in steady state will
+    // be less than this number when ((.spec.completions - .status.successful) < .spec.maxParallelism),
+    // i.e. when the work left to do is less than max parallelism.
+    MaxParallelism *int
+
+    // Completions specifies the desired number of successfully finished pods the
+    // job should be run with. Defaults to 1.
+    Completions *int
+
+    // Selector is a label query over pods that should match the pod count.
+    Selector map[string]string
+
+    // Template is the object that describes the pod that will be created when
+    // executing a job.
+    Template *PodTemplateSpec
+}
+```
+
+`JobStatus` structure is defined to contain informations about pods executing
+specified job.  The structure holds information about pods currently executing
+the job.
+
+```go
+// JobStatus represents the current state of a Job.
+type JobStatus struct {
+    Conditions []JobCondition
+
+    // CreationTime represents time when the job was created
+    CreationTime util.Time
+
+    // StartTime represents time when the job was started
+    StartTime util.Time
+
+    // CompletionTime represents time when the job was completed
+    CompletionTime util.Time
+
+    // Active is the number of actively running pods.
+    Active int
+
+    // Successful is the number of pods successfully completed their job.
+    Successful int
+
+    // Unsuccessful is the number of pods failures, this applies only to jobs
+    // created with RestartPolicyNever, otherwise this value will always be 0.
+    Unsuccessful int
+}
+
+type JobConditionType string
+
+// These are valid conditions of a job.
+const (
+    // JobSucceeded means the job has successfully completed its execution.
+    JobSucceeded JobConditionType = "Complete"
+)
+
+// JobCondition describes current state of a job.
+type JobCondition struct {
+    Type               JobConditionType
+    Status             ConditionStatus
+    LastHeartbeatTime  util.Time
+    LastTransitionTime util.Time
+    Reason             string
+    Message            string
+}
+```
+
+## Events
+Job controller will be emitting the following events:
+* JobStart
+* JobFinish
+
+## Future evolution
```

Comment:
I think `kubectl describe job` could join a list of the Pods created by the Job controller with events from those pods, which removes the need to keep the Pod themselves.